12_MiscellaneousDesign.md

```diff
@@ -64,6 +64,7 @@ class MyClass<T>
     }
 }
 ```
+ **Note:** In this case, classes of generic types are children of the SomeClass, so replacing T with the SomeClass (MyClass<SomeClass>) to make a cast is not allowed.
```

Comment:
This note seems confusing. I suggest updating description above to sth like this:
`Instead of casting to and from the object type in generic types or methods, use where constraints or the as operator to specify the exact characteristics of the generic parameter (assume you can not get rid of the generic parameter).`
instead of providing some specific situations

Reply:
I wonted to make the example more real. But it seems that your variant will be easier to perceive.